src/runtime/classmanager.cs

```diff
@@ -481,5 +474,28 @@ internal ClassInfo()
             members = new Hashtable();
             indexer = null;
         }
+
+        ~ClassInfo()
+        {
+            Dispose();
```

Comment:
It's not safe to call such a GIL needed method in GC thread, besides, its reference count already balanced, thus the `Dispose` is not necessary.
https://github.com/amos402/pythonnet/blob/2b7bcacb63b25fb760e38bc558e40cedd3351933/src/runtime/classmanager.cs#L224

Reply:
Then the fix is to document that ClassInfo should not be used outside of ClassManager and that the caller is responsible to decrease the reference count of the members.

Reply:
It didn't fix any things, it made bugs.
Check here:
https://github.com/amos402/pythonnet/blob/2b7bcacb63b25fb760e38bc558e40cedd3351933/src/runtime/classmanager.cs#L217-L225

Reply:
ClassInfo is merely a temporary object for carrying the data. Of course, you can clear the members after using it although it won't influence the usages.
And the type of `member` can change into a generic type, it's some old code so it didn't use generic type. Well, but seems it's not necessary to change it here.

Reply:
I will revert this change and add documentation instead